### ISSUE TITLE
Azure Cloud Shell to connect to Sovereign Clouds

### ISSUE THREAD

**Issue by davemedvitz** [microsoft/terminal#3597]
The Azure Cloud Shell component needs to be able to connect to Azure Clouds other than the default Commercial cloud.  Currently the relevant URL endpoints are hard coded to Commercial.

**Comment by DHowett-MSFT** [comment]
Yeah, fair point. Marking up for the backlog. :smile: Thanks

**Comment by jmcarthurcorr** [comment]
Giving a +1 to this issue. Logging in with portal.azure.us tenant gives the dreaded

> AADSTS50011: The reply url specified in the request does not match the reply urls configured for the application: '245e1dee-74ef-4257-a8c8-8208296e1dfd'.

**Comment by petkahl** [comment]
I see from the duplicate issue that this is waiting on compliance as much as code. Is there anything those of us on the outside could do to help (write code and queue up a PR, etc?) or is it really just a matter of waiting for the compliance/legal part to be handled?

**Comment by DHowett** [comment]
This is all compliance stuff; the real work involved is in the MS first-party identity portal where we have to onboard to the sovereign clouds. That should be quick once we get signoff on security and accessibility for our service :smile:

**Comment by davemedvitz** [comment]
Do know if there is progress being made on the compliance pieces?

**Comment by DHowett** [comment]
Unfortunately, I don't have any news to share on that front.

**Comment by hphungnal** [comment]
Hello, just checking in to see if there is any update on this issue? Would love for it to work with https://microsoft.com/deviceloginus
Thanks!

**Comment by benatsb** [comment]
Hello, any updates on this capability?

**Comment by zadjii-msft** [comment]
Nope. We'll make sure to update this thread when there is. In the meantime, might I recommend the Subscribe button?
![image](https://user-images.githubusercontent.com/18356694/91237459-5cbb0c80-e700-11ea-9347-b9b1ec2813b1.png)
That way you'll be notified of any updates to this thread, without needlessly pinging everyone on this thread ☺️

**Comment by zadjii-msft** [comment]
The crazy idea from sync: Rather than onboarding the Terminal's AppID onto each cloud ourselves, we enable configuration options s.t. the user can specify their own AppID. That way, each entity could onboard and approve the Terminal themselves, synthesize their own appid for the cloud shell, and then users would just add that string to their settings file. 

It's more burden on the users, for sure, but would allow each individual cloud the ability to control access to the azure cloud shell themselves.

**Comment by jamesseiwert** [comment]
Is there any update on this? I was hoping with the cross cloud access now this would be resolved. I also like the idea from @zadjii-msft about enabling configuration options as a secondary idea. Most microsoft apps can auth can handle both commercial and goc clouds now
